### PR TITLE
non state module fields

### PR DESCRIPTION
**Pull request by eyusufatik**
# Description
shared memory between hooks and module calls 

no need for pending_transactions and block_env

TODOs:
- [x] make block_env #[memory]

### CODE REVIEW

**Comment by eyusufatik** [comment]
I suggest only reviewing [0d04e0a](https://github.com/chainwayxyz/citrea/pull/1110/commits/0d04e0a1be960e49b432185645378286504a2a78)

File: crates/evm/src/call.rs [hunk 1]
```diff
@@ -80,7 +80,9 @@ impl<C: sov_modules_api::Context> Evm<C> {
 
         let mut cumulative_gas_used = 0;
         let mut log_index_start = 0;
-        if let Some(tx) = self.pending_transactions.last(working_set) {
+
+        if !self.pending_transactions.is_empty() {
```
Comment (kpp) [inline crates/evm/src/call.rs:84]: If they are not empty we are doing smth wrong

File: crates/evm/src/call.rs [hunk 2]
```diff
@@ -149,7 +150,9 @@ impl<C: sov_modules_api::Context> Evm<C> {
         let block_number = block_env.number;
         let mut cumulative_gas_used = 0;
         let mut log_index_start = 0;
-        if let Some(tx) = self.pending_transactions.last(working_set) {
+
+        if !self.pending_transactions.is_empty() {
```
Comment (kpp) [inline crates/evm/src/call.rs:154]: if let Some(tx) = self.pending_transactions.last()

File: crates/evm/src/call.rs [hunk 3]
```diff
@@ -149,7 +150,9 @@ impl<C: sov_modules_api::Context> Evm<C> {
...
+            let tx = self.pending_transactions.last().unwrap();
```
Comment (kpp) [inline crates/evm/src/call.rs:155]:
```suggestion
        if let Some(tx) = self.pending_transactions.last() {
```

File: crates/evm/src/call.rs (hunk 1 above)
Comment (eyusufatik) [reply crates/evm/src/call.rs:79]: remember when you were implementing sys txs and I asked you why you did some stuff in a certain way you said maybe we'll run sys txs at the end of the blocks in the future.

:)

**Comment by eyusufatik** [comment]
[b59d67d](https://github.com/chainwayxyz/citrea/pull/1110/commits/b59d67d07db28a448c99e4798a7183f348720718)
[da3bd19](https://github.com/chainwayxyz/citrea/pull/1110/commits/da3bd19315c565b2cd3dc2245bc8c5b5c83e6f39)

these two are also worty of review

File: crates/evm/src/call.rs (hunk 1 above)
Comment (kpp) [reply crates/evm/src/call.rs:79]: I was wrong =)

Comment (eyusufatik) [reply crates/evm/src/call.rs:79]: I can make another pr that makes sure execute sys txs is called at the beginning. it will also save us some cycles

Comment (kpp) [reply crates/evm/src/call.rs:79]: Yes

**Pull request merged**